parquet/src/main/java/org/apache/iceberg/parquet/ParquetReadSupport.java

```diff
@@ -55,9 +58,16 @@ public ReadContext init(Configuration configuration, Map<String, String> keyValu
     // matching to the file's columns by full path, so this must select columns by using the path
     // in the file's schema.
 
-    MessageType projection = ParquetSchemaUtil.hasIds(fileSchema) ?
-        ParquetSchemaUtil.pruneColumns(fileSchema, expectedSchema) :
-        ParquetSchemaUtil.pruneColumnsFallback(fileSchema, expectedSchema);
+    MessageType projection;
+    if (ParquetSchemaUtil.hasIds(fileSchema)) {
+      projection = ParquetSchemaUtil.pruneColumns(fileSchema, expectedSchema);
+    } else {
+      if (nameMapping != null) {
+        projection = ParquetSchemaUtil.pruneColumnsByName(fileSchema, expectedSchema, nameMapping);
```

Comment:
When a name mapping is used, column pruning should happen using the normal `pruneColumns`. The reason why the other fallback path uses `pruneColumnsFallback` is because ordinal IDs can only be assigned to the top-level columns and so projection must only use top-level column IDs. That's not a limitation of name mapping, so we can use normal pruning once the IDs are filled in.

Reply:
I got your point now. Will remove `pruneColumnsByName`.